_docs-sources/pipelines/maintain/drift-detection.md

```diff
@@ -0,0 +1,110 @@
+
+  jobs:
+    GruntworkPipelines:
+      uses: gruntwork-io/pipelines-workflows/.github/workflows/pipelines-drift-detection.yml@v2
+      secrets:
+        PIPELINES_READ_TOKEN: ${{ secrets.PIPELINES_READ_TOKEN }}
+        # Uncomment the following line for your repository:
+        # infrastructure-live-root:
+        # PR_CREATE_TOKEN: ${{ secrets.INFRA_ROOT_WRITE_TOKEN }}
+        # infrastructure-live-access-control:
+        # PR_CREATE_TOKEN: ${{ secrets.ACCESS_CONTROL_WRITE_TOKEN }}
+        # delegated repositories:
+        # PR_CREATE_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  ```
+
+</details>
+3. Uncomment the appropriate line in the permissions block for pull-request permissions
+4. Uncomment the appropriate line in the secrets block for PR_CREATE_TOKEN
+5. For delegated repositories - ensure that **Allow GitHub Actions to create and approve pull requests** is checked in the repository settings.
+
+![Allow Actions To Create Pull Requests](/img/pipelines/maintain/drift_detection_allow_pr_create.png)
+
+
+## Detecting Drift
+
+Pipelines Drift Detection can be run on a schedule or manually.
+
+### Running on a schedule
+
+To enable running on a schedule, uncomment the schedule block containing `- cron: '0 0 * * 1-5'` in `.github/workflows/pipelines-drift-detection.yml`. The default schedule runs at 00:00UTC Monday through Friday. You can increase or decrease the frequency that the schedule runs using [crontab syntax](https://crontab.guru/#0_0_*_*_1-5).
+
+Each time Drift Detection runs and detects drift it will open a Pull Request in your repository. If there is an existing Drift Detection Pull Request that has not been merged it will be replaced.
+
+:::note
+Running Pipelines Drift Detection too frequently can easily eat through your GitHub Action minutes. We recommend starting with a low frequency and increasing only when you are comfortable with the usage.
+:::
+
+### Running manually
+
+Pipelines Drift Detection can be run manually by navigating to Actions in your GitHub repository, selecting Pipelines Drift Detection from the left hand menu, and then selecting Run Workflow.
+
+## Customizing the Drift Detection Branch Name
+
+By default Pipelines will create a Pull Request on the branch `drift-detection`. You can customize the branch name used for the Drift Detection Pull Request by adding a branch-name parameter to `./gitub/workflows/pipelines-drift-detection.yml`
+
+E.g.
+
+```yml
+  jobs:
+    GruntworkPipelines:
+      uses: gruntwork-io/pipelines-workflows/.github/workflows/pipelines-drift-detection.yml@v2
+      with:
+        branch-name: "customized-branch-name"
+```
+
+## Resolving Drift
+
+Drift can be resolved by either applying the commited IaC from your repository, or modifying modules until they reflect the infrastructure state in the cloud.
+
+### Merging The Pull Request
+
+Merging the Pull Request will trigger a `terragrunt apply` on the drifted modules. In most cases this will resolve the drift.
```

Comment:
In what cases would it not resolve the drift?

Reply:
I figured there's probably flaky modules out there - better to remove?